### PR TITLE
Feat/ephemeral storage limit

### PR DESCRIPTION
**Pull request by clementnuss**
add a new ephemeral storage limit constraint, heavily inspired by the `containerlimits` constraint.

> Requires containers to have an ephemeral storage limit set and constrains the limit to be within the specified maximum values

### CODE REVIEW

**Comment by ritazh** [comment]
Thank you for the PR @clementnuss! Apologies for the delay. Can you PTAL https://open-policy-agent.github.io/gatekeeper-library/website/#development to run few commands as we have updated few things to autogenerate changes for the website and artifacthub.

**Comment by clementnuss** [comment]
hi @ritazh 

I've updated my branch by following the latest instructions. can you review this PR ?

**Comment by clementnuss** [comment]
@maxsmythe @ritazh 

I think this PR would be beneficial to the gatekeeper library, could it be merged? I'ts been on hold for quite a long time now 🤔

File: library/general/ephemeralstoragelimit/suite.yaml
Comment (maxsmythe) [inline library/general/ephemeralstoragelimit/suite.yaml:11]: Could we add in a test case for init containers?

**Review by maxsmythe — COMMENTED** [review]
Sorry this slipped through the cracks. One request for an added test to suite.yaml, other than that, LGTM!

File: library/general/ephemeralstoragelimit/suite.yaml
Comment (clementnuss) [reply library/general/ephemeralstoragelimit/suite.yaml:11]: sure, good idea!

**Comment by clementnuss** [comment]
I've added 2 test cases for initContainer: 1 for allowed initContainer and another one with a 1Pi ephemeral storage limit for disallow 👍🏻

**Review by maxsmythe — APPROVED** [review]
Thanks!

LGTM

**Review by sozercan — APPROVED** [review]
Sorry about the delay and thank you! LGTM pending tests

**Comment by clementnuss** [comment]
hey @sozercan 

Could you let the CI pipeline run again ? I've updated website docs and artifactshub artifacts as I had forgotten to update that when I added the two tests.

pending tests

**Pull request merged**
